system/src/system_cloud_internal.cpp

```diff
@@ -1053,6 +1054,15 @@ int Spark_Handshake(bool presence_announce)
     LOG(INFO,"Starting handshake: presense_announce=%d", presence_announce);
     bool session_resumed = false;
     int err = spark_protocol_handshake(sp);
+
+    // Adding a delay only for platforms Boron and BSom, because older cell versions of
+    // Boron R410M modems crash when handshake messages are sent without gap
+#if HAL_PLATFORM_MUXER_MAY_NEED_DELAY_IN_TX
+        if (platform_current_ncp_identifier() == PLATFORM_NCP_SARA_R410) {
+            HAL_Delay_Milliseconds(1000);
```

Comment:
Do we perhaps want to query the firmware version here as well? I think it should be alright to directly query SaraNcpClient here and perhaps make the timeout come from there as well to avoid using a magic value. So something along the lines of:
```
#if HAL_PLATFORM_MUXER_MAY_NEED_DELAY_IN_TX
auto timeout = cellularNetworkManager()->ncpClient()->mayNeedTxDelay(); // don't judge the name :)
if (timeout > 0) {
    HAL_Delay_Milliseconds(timeout);
}
#endif // HAL_PLATFORM_MUXER_MAY_NEED_DELAY_IN_TX
```